pkg/http/server.go

```diff
@@ -60,6 +67,27 @@ type server struct {
 	tlsListener net.Listener
 }
 
+func fdToListener(bindAddress string) (net.Listener, error) {
+	fd, err := strconv.Atoi(bindAddress)
+	if err != nil {
+		return nil, fmt.Errorf("listen failed: fd with name is not implemented yet.")
+	}
+
+	files := activation.Files(true)
+	if fd < 3 || len(files) < fd-3 {
+		return nil, fmt.Errorf("listen failed: fd (%d) out of bounds (len %d)", fd, len(files))
```

Comment:
What does out of bounds really mean here? What is the significance of the number 3 here? I'd appreciate some code comments to explain this I think

Reply:
When the process starts it recieves by default the following file descriptors:
+ 0 - stdin
+ 1 - stdout
+ 2 - stderr

When running with systemd-socket-activate it also receives:
+ 3 - first file descriptor that systemd sends along

Which means that if files=[3] and FD=4, or FD=2, it shall fail since it's not a valid file descriptor. It must be greater or equal to 3.

Reply:
I made a better comment about this around the listenFdsStart constant

Reply:
@JoelSpeed enough of a comment?